### DISCUSSION TITLE
repeater wifi with ESP32 and micropython : doable ?

### DISCUSSION THREAD

**Discussion by picpic020960** [micropython/micropython#11907] [ESP32]
Bonjour ,
I have tryed this firmware (not micropython)
 [https://github.com/martin-ger/esp32_nat_router]
(sorry : use ctrl-C and ctrl-V , I don't know why ...) 
works fine.
Is it possible with micropython script ?
Thank's for help/links
Regards

**Comment by peterhinch** [comment]
That link is 404.

**Comment by picpic020960** [comment]
(sorry : use ctrl-C and ctrl-V , I don't know why ...)

**Comment by karfas** [comment]
In the moment, there is no way in micropython to instruct LWIP to do NAT, routing or other low-level network stuff, I think.

What exactly do you mean by "Is it possible with micropython script" ?

It *might* be possible to create something similar using RAW sockets, but I assume the performance will be horrible compared to a C solution, leaving alone security implications (e.g. recreate some functions of a battle-tested TCP stack in micropython).
